CorsixTH/Lua/dialogs/jukebox.lua

```diff
@@ -176,3 +176,9 @@ function UIJukebox:draw(canvas, x, y)
     end
   end
 end
+
+function UIJukebox:afterLoad()
+  if not (self.app.config.audio and self.app:isAudioEnabled()) then
+    self:close()
+  end
+end
```

Comment:
Apparently you need to call the base-class too, see eg `dialogs/adviser.lua`. You also need to pass the `old` and `new` values.

Not sure if it needs an `old` test like the other `afterLoad` function, I guess it won't hurt much not to.